### DISCUSSION TITLE
How to map multiple extension filenames to lexer?

### DISCUSSION THREAD

**Discussion by michaeldurland** [orbitalquark/textadept#113] [Q&A]
How do I map a filename with multiple extensions to a specific lexer?
For example, let's say I have a filename called "script.foo.bar" and I want to have it use the "bash" lexer.
I tried adding this to my init.lua, but it didn't seem to work:
textadept.file_types.extensions['foo.bar'] = 'bash'
Thanks!

**Comment by orbitalquark** [comment]
Unfortunately, this is not possible.

For reference: https://github.com/orbitalquark/textadept/blob/83a49ca00580a3d5a987a9132ac4b0483a57bf4d/modules/textadept/file_types.lua#L52

**Reply by michaeldurland** [reply]
Thank you for the pointer to the code that detects the file extension. That led me to write my own event handler to set the lexer for my specific double extension:
```lua
events.connect(events.FILE_OPENED, function(filename)
    if filename:match('.*.foo.bar') then
        buffer:set_lexer('bash')
    end
end)
```
It might not be perfect, but it seems to do the job for my use case.
